packages/backend-api/install/tests/ShopBundle/Smoke/BackendApiCreateProductTest.php

```diff
@@ -0,0 +1,215 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\ShopBundle\Smoke;
+
+use Ramsey\Uuid\Uuid;
+use Tests\ShopBundle\Test\OauthTestCase;
+
+/**
+ * This test must not extend TransactionFunctionalTestCase because it mustn't use transaction
+ * If it is run in transaction, tokens don't work and test fails
+ */
+class BackendApiCreateProductTest extends OauthTestCase
+{
+    public function testCreatePostProduct(): void
+    {
+        $product = $this->getValidProduct();
+        $response = $this->runOauthRequest('POST', '/api/v1/products', $product);
+
+        $this->assertSame(201, $response->getStatusCode());
+
+        $location = $response->headers->get('Location');
+        $this->assertProductLocation($location);
+
+        $uuid = $this->extractUuid($location);
+
+        $foundProduct = $this->getProduct($uuid);
+        $this->assertSame($product, $foundProduct);
+    }
+
+    public function testCreateProductWithSpecifiedUuid(): void
+    {
+        $product = $this->getValidProduct();
+        $uuid = Uuid::uuid4()->toString();
+        $product['uuid'] = $uuid;
+        $response = $this->runOauthRequest('POST', '/api/v1/products', $product);
+
+        $this->assertSame(201, $response->getStatusCode());
+
+        $location = $response->headers->get('Location');
+        $this->assertProductLocation($location);
+
+        $extractedUuid = $this->extractUuid($location);
+        $this->assertSame($uuid, $extractedUuid);
+
+        $foundProduct = $this->getProduct($extractedUuid);
+        unset($product['uuid']);
```

Comment:
You unset uuid here and also in `getProduct` method. But you can compare products incl. UUIDs, right?

Reply:
Well, I removed `unset` and had to fix tests with POST without UUID. The outcome is -5 and +5 lines :-D I'll keep it after the change